djes/search.py

```diff
@@ -100,3 +100,73 @@ def execute(self):
                                                 **self._params))
 
         return self._executed
+
+
+class SearchParty(object):
+    """
+    Generator for multiple searches that allows for searches to iterate through the results of
+    multiple queries, while providing explicit logic for indexing.
+    """
+    def __init__(self, *args, **kwargs):
+        self.results = []
+        self.searches = {}
+        self.primary_search = None
+
+    def __getitem__(self, n):
+        if n < len(self.results):
+            return self.results[n]
+        elif n - len(self.results) > 1:
+            for i in range(len(self.results), n):
+                self[i]
+        search = self.get_search(n)
+        self.results.append(self.get_result(search))
+        return self.results[n]
+
+    def get_search(self, n):
+        for search, config in self.searches.items():
+            search_ranges = config.get("ranges")
+            if search_ranges:
+                for search_range in search_ranges:
+                    if not search_range:
+                        continue
+                    elif n >= search_range[0] and n < search_range[1]:
+                        return search
+        return self.primary_search
+
+    def get_result(self, search):
+        count = self.searches[search]["count"]
+        result = search[count]
+        self.searches[search]["count"] += 1
+        return result
+
+    def register_search(self, search, search_range=None, primary=False):
```

Comment:
Wouldn't this undermine the point of the PR? It seems like doing that we would just write a generator for every off case instead of having an object manage them.